apps/postgres-new/components/app-provider.tsx

```diff
@@ -138,7 +138,17 @@ export default function AppProvider({ children }: AppProps) {
         if (isStartupMessage(message)) {
           const parameters = parseStartupMessage(message)
           if ('client_ip' in parameters) {
-            setConnectedClientIp(parameters.client_ip === '' ? null : parameters.client_ip)
+            // client disconnected
+            if (parameters.client_ip === '') {
+              setConnectedClientIp(null)
+              // we ensure we're not in a transaction block first
+              await db.sql`rollback;`.catch()
+              // we clean the session state, see: https://www.pgbouncer.org/faq.html#how-to-use-prepared-statements-with-session-pooling
+              // we do this to avoid having old prepared statements in the session
+              await db.sql`discard all;`
```

Comment:
Added after working on making `pg_dump` works. Running `pg_dump` twice showed that we kept old prepared statements in the session for the next connected client, which ended up erroring as the next pg_dump creates the same prepare statement.

Reply:
I suppose we should mention `rollback` + `discard all` in the pg-gateway pglite examples too? I imagine this would apply to any pg+gateway + PGlite stack.